source/android/logging.c

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <android/log.h>
+
+#include <aws/common/logging.h>
+#include <aws/common/string.h>
+
+#include <stdarg.h>
+
+#define LOGCAT_MAX_BUFFER_SIZE 1024
+
+struct logcat_format_data {
+    char *buffer;
+    size_t bytes_written;
+    size_t total_length;
+    const char *format;
+    const char *subject;
+};
+
+static size_t s_advance_and_clamp_index(size_t current_index, int amount, size_t maximum) {
+    size_t next_index = current_index + amount;
+    if (next_index > maximum) {
+        next_index = maximum;
+    }
+
+    return next_index;
+}
+
+/* Override this for Android, as time and log level are taken care of by logcat */
+static int s_logcat_format(struct logcat_format_data *formatting_data, va_list args) {
```

Comment:
I feel like there could be a little bit of refactoring shared functionality between this and the default via a private header or something.  Might be more apparent as we add other platform logger implementations.

Reply:
For sure. We could probably just have a flagset or something per platform that indicates which parts of the format are required.